diff --git a/aichallenge2023-sim/autorun_server.sh b/aichallenge2023-sim/autorun_server.sh
index b74983a..e1697bc 100644
--- a/aichallenge2023-sim/autorun_server.sh
+++ b/aichallenge2023-sim/autorun_server.sh
@@ -49,8 +49,19 @@ function run_awsim(){
     sleep 5
     echo "-- run AWSIM... -->"
     echo "CMD: ${AWSIM_EXEC_COMMAND}"
-    gnome-terminal -- bash -c "${AWSIM_EXEC_COMMAND}" &
-    sleep 15
+    #gnome-terminal -- bash -c "${AWSIM_EXEC_COMMAND}" &
+    #sleep 15
+    for ((i=0; i<20; i++));
+    do
+	gnome-terminal -- bash -c "${AWSIM_EXEC_COMMAND}" &
+	sleep 15
+	PROCESS_CNT=`ps -aux | grep "docker run --rm -it --name ${AWSIM_ROCKER_NAME}" | wc -l`
+	if [ ${PROCESS_CNT} >= 2 ]; then
+	    break
+	fi
+	echo "no process ${AUTOWARE_ROCKER_NAME}, retry.."
+    done
+
     return
 }
 
